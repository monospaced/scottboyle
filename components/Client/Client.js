import React from "react";

const Client = ({ client: { link, title } }) => {
  return (
    <p>
      <strong>For:</strong> {!link ? title : <a href={link}>{title}</a>}
    </p>
  );
};

export default Client;
