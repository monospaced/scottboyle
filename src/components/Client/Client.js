import React from "react";

const Client = ({ client: { link, title } }) => {
  return title ? (
    <p>
      <strong>For:</strong> {!link ? title : <a href={link}>{title}</a>}
    </p>
  ) : null;
};

export default Client;
