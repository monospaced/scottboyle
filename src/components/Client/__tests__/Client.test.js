import React from "react";
import { render } from "enzyme";

import Client from "../Client.js";

describe("Client component", () => {
  it("should render correctly", () => {
    const component = render(<Client client={{}} />);
    expect(component).toMatchSnapshot();
  });

  it("should render a name", () => {
    const component = render(<Client client={{ title: "Client" }} />);
    expect(component).toMatchSnapshot();
  });

  it("should render a link", () => {
    const component = render(
      <Client client={{ title: "Client", link: "https://client.com" }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
