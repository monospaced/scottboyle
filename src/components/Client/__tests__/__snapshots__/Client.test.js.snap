// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client component should render a link 1`] = `
<p>
  <strong>
    For:
  </strong>
   
  <a
    href="https://client.com"
  >
    Client
  </a>
</p>
`;

exports[`Client component should render a name 1`] = `
<p>
  <strong>
    For:
  </strong>
   Client
</p>
`;

exports[`Client component should render correctly 1`] = `null`;
